### PR TITLE
[autoscaler] Fix AWS Autoscaler Tests

### PR DESCRIPTION
**Pull request by pdames**
## Why are these changes needed?

Fix broken AWS Autoscaler CloudWatch unit test stubs and remove broken/outdated EC2 security group and network interface tests.

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed.
- [X] I've made sure the tests are passing. Note that there might be a few flaky tests.
- Testing Strategy
   - [X] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: python/ray/autoscaler/_private/aws/node_provider.py [hunk 1]
```diff
@@ -420,13 +420,15 @@ def _create_node(self, node_config, tags, count):
...
                         exc)
-        return created_nodes_dict
 
         # TODO: Idempotently correct CloudWatch setup errors on cached nodes?
         node_ids = [n.id for n in created]
-        CloudwatchHelper(self.provider_config, node_ids, self.cluster_name).\
+        CloudwatchHelper(self.provider_config, node_ids, self.cluster_name). \
             setup_from_config()
 
+        return created_nodes_dict
```
Comment (dccorona) [inline python/ray/autoscaler/_private/aws/node_provider.py:429]: Did the code just not run before, moving return makes it actually happen?

**Review by dccorona — APPROVED** [review]

**Comment by annaluo676** [comment]
Code changes LGTM. 

p.s.: I noticed several task failures in Travis CI. Some of them can also be seen in Ray's official repo's PRs (e.g., `PYTHON=3.6 SMALL_AND_LARGE_TESTS=1`) whereas some others are not commonly seen (e.g., `LINT=1`). After reading the [understanding-ci-test-jobs](https://docs.ray.io/en/master/getting-involved.html#understanding-ci-test-jobs) section it doesn't seem to be blocking issue since this PR is within the scope of aws autoscaler.

**Review by annaluo676 — APPROVED** [review]

File: python/ray/autoscaler/_private/aws/node_provider.py (hunk 1 above)
Comment (pdames) [reply python/ray/autoscaler/_private/aws/node_provider.py:429]: Right - this was a side-effect of a bad merge.

**Comment by pdames** [comment]
> Code changes LGTM.
> 
> p.s.: I noticed several task failures in Travis CI. Some of them can also be seen in Ray's official repo's PRs (e.g., `PYTHON=3.6 SMALL_AND_LARGE_TESTS=1`) whereas some others are not commonly seen (e.g., `LINT=1`). After reading the [understanding-ci-test-jobs](https://docs.ray.io/en/master/getting-involved.html#understanding-ci-test-jobs) section it doesn't seem to be blocking issue since this PR is within the scope of aws autoscaler.

I noticed the same - it doesn't look like our changes have introduced any new failures, but it would be nice to at least get the LINT job working for all branches in the repo going forward.

**Pull request merged**